python_modules/dagster/dagster_tests/definitions_tests/declarative_automation_tests/daemon_tests/test_e2e.py

```diff
@@ -383,3 +410,114 @@ def test_non_subsettable_check() -> None:
+    ) as context, get_threadpool_executor() as executor:
+        asset_graph = context.create_request_context().asset_graph
+
+        # all start off missing, should be requested
+        time = get_current_datetime()
+        with freeze_time(time):
+            _execute_ticks(context, executor)
+            backfills = _get_backfills_for_latest_ticks(context)
+            assert len(backfills) == 1
+            subsets_by_key = _get_subsets_by_key(backfills[0], asset_graph)
+            assert subsets_by_key.keys() == {
+                AssetKey("A"),
+                AssetKey("B"),
+                AssetKey("C"),
+                AssetKey("D"),
+                AssetKey("E"),
+            }
+
+            assert subsets_by_key[AssetKey("A")].size == 1
+            assert subsets_by_key[AssetKey("B")].size == 3
+            assert subsets_by_key[AssetKey("C")].size == 3
+            assert subsets_by_key[AssetKey("D")].size == 3
+            assert subsets_by_key[AssetKey("E")].size == 1
+
+            # don't create runs
+            runs = _get_runs_for_latest_ticks(context)
+            assert len(runs) == 0
+
+        time += datetime.timedelta(seconds=30)
+        with freeze_time(time):
+            # second tick, don't kick off again
+            _execute_ticks(context, executor)
+            backfills = _get_backfills_for_latest_ticks(context)
+            assert len(backfills) == 0
+            # still don't create runs
+            runs = _get_runs_for_latest_ticks(context)
+            assert len(runs) == 0
+
+
+def test_backfill_with_runs_and_checks() -> None:
+    with get_workspace_request_context(
+        ["backfill_with_runs_and_checks"]
+    ) as context, get_threadpool_executor() as executor:
+        asset_graph = context.create_request_context().asset_graph
+
+        # report materializations for 2/3 of the partitions, resulting in only one
+        # partition needing to be requested
+        context.instance.report_runless_asset_event(AssetMaterialization("run2", partition="x"))
+        context.instance.report_runless_asset_event(AssetMaterialization("run2", partition="y"))
+
+        # all start off missing, should be requested
+        time = get_current_datetime()
+        with freeze_time(time):
+            _execute_ticks(context, executor)
+            # create a backfill for the part of the graph that has multiple partitions
+            # required
+            backfills = _get_backfills_for_latest_ticks(context)
+            assert len(backfills) == 1
+            subsets_by_key = _get_subsets_by_key(backfills[0], asset_graph)
+            assert subsets_by_key.keys() == {
+                AssetKey("backfillA"),
+                AssetKey("backfillB"),
+                AssetKey("backfillC"),
+            }
+
+            assert subsets_by_key[AssetKey("backfillA")].size == 1
+            assert subsets_by_key[AssetKey("backfillB")].size == 3
+            assert subsets_by_key[AssetKey("backfillC")].size == 3
```

Comment:
follow up for [1] - my read is that the `outside` asset checks are supposed to be run here?

Reply:
The goal is actually for these checks to *not* be executed alongside their respective assets, as it is not required for them to be executed with their assets.

It's analogous in some ways to having a multi_asset with "can_subset=True" -- we shouldn't assume that a request to materialize an asset with a non-required check implies that we should execute that check as well.

On the flip-side, for checks which *are* required (i.e. defined within the asset), we need to make sure that those are properly requested, otherwise we'll hit an error if we request an "invalid" run.

Reply:
ok that makes sense. I was missing the info that the asset checks defined outside the asset function are by definition not required